civitai-downloader-v2/src/api/main.py

```
@@ -0,0 +1,518 @@
+            limit=search_params.limit,
+            page=search_params.page
+        )
+        
+        # Store model metadata in database for caching
+        for model in result.get("models", []):
+            database_service.store_model_metadata(model)
+        
+        return result
+        
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Search failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)